compiler/rustc_lint/src/unnecessary_block.rs

```diff
@@ -0,0 +1,45 @@
+use crate::lints::UnnecessaryBlock as UnnecessaryBlockErr;
+use crate::{LateContext, LateLintPass, LintContext};
+use rustc_hir::{BlockCheckMode, ExprKind, Stmt, StmtKind};
+
+declare_lint! {
+    /// The `unnecessary_block` lint checks for unnecessary block.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// fn main() {
+    ///     {}
+    /// }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// Having unnecessary block is almost always a mistake.
```

Comment:
Minor grammatical nitpick:
```suggestion
    /// Having an unnecessary empty block is almost always a mistake.
```